ojd_daps_skills/analysis/hyperparameter_analysis.py

```diff
@@ -0,0 +1,179 @@
+# ---
+# jupyter:
+#   jupytext:
+#     cell_metadata_filter: -all
+#     comment_magics: true
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.13.8
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %% [markdown]
+# ## Hyperparameter analysis
+#
+# The NER model was trained several times with different hyperparameters (number of iterations, learning rate, drop out rate) to find which parameters gave the best results. The results of the models were saved out and are analysed here.
+#
+# ### [Experiment 1](#exp_1)
+# 30 iterations, 27 different random combos learn rate and drop out rates, no evaluation results
+# - Do losses stabilise after 30 iterations?
+# - Which hyperparameters give the best final loss?
+#
+# ### [Experiment 2](#exp_2)
+# 10 iterations, 20 different random combos learn rate and drop out rates, evaluation results
+# - Does the F1 score for all and just skill entities change much with the hyperparams?
+# - Which hyperparameters give the best final loss and model scores?
+#
+# #### Download the data:
+# ```
+# aws s3 cp s3://open-jobs-lake/escoe_extension/outputs/quality_analysis/results_12_20220824.json outputs/data/skill_ner/parameter_experiments/results_12_20220824.json
+# aws s3 cp s3://open-jobs-lake/escoe_extension/outputs/quality_analysis/results_10_20220824.json outputs/data/skill_ner/parameter_experiments/results_10_20220824.json
+# aws s3 cp s3://open-jobs-lake/escoe_extension/outputs/quality_analysis/results_5_20220824.json outputs/data/skill_ner/parameter_experiments/results_5_20220824.json
+# aws s3 cp s3://open-jobs-lake/escoe_extension/outputs/quality_analysis/results_20_20220824_eval.json outputs/data/skill_ner/parameter_experiments/results_20_20220824_eval.json
+#
+# ```
+
+# %%
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+
+import json
+import ast
+import random
+
+from ojd_daps_skills import PROJECT_DIR
+
+# %% [markdown]
+# ## Experiment 1<a id='exp_1'></a>
+
+# %%
+data = []
+with open(
```

Comment:
not a bug but we may want to add these parameter experiment .jsons to the repo so people can run this with the relevant files

Reply:
ignore me